File: Java/deepspace/src/main/java/deepspace/EnemyStarShip.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author fjsae
 */
public class EnemyStarShip {
    
    private float ammoPower;
    private String name;
    private float shieldPower;
    private Loot loot;
    private Damage damage;
    
    public float getAmmoPower() {
        return ammoPower;
    }

    public String getName() {
        return name;
    }

    public float getShieldPower() {
        return shieldPower;
    }

    public Loot getLoot() {
        return loot;
    }
    
    public Damage getDamage(){
        return damage;
    }
    
    EnemyStarShip(String n, float s, float a, Loot l, Damage d) {
        this.name = n;
        this.ammoPower = a;
        this.shieldPower = s;
        this.loot = l;
        this.damage = d;
        
    }
    
    EnemyStarShip(EnemyStarShip e){
        this.ammoPower = e.ammoPower;
        this.name = e.name;
        this.shieldPower = e.shieldPower;
        this.loot = e.loot;
        
    }
    
    public float fire(){
        return this.ammoPower;
    }
    
    public float protection(){
        return this.shieldPower;
    }
    
    public ShotResult receiveshot(float shot){
        ShotResult ret = ShotResult.RESIST;
        if(this.shieldPower < shot){
            ret  = ShotResult.DONOTRESIST;
        }
        return ret;
    }
    
    public EnemyToUI getUIversion(){
        return new EnemyToUI(this);
    }
        
}

File: Ruby/EnemyStarShip.rb
#encoding:utf-8

require_relative 'EnemyToUI'
require_relative 'ShotResult'

module Deepspace

    class EnemyStarShip

        def initialize(n,a,s,l,d)
            @name = n 
            @ammoPower = a
            @shieldPower = s
            @loot = l
            @damage = d 
        end

        def to_s
            "+Enemy NAME: #{@name}.\n \tAmmoPower = #{@ammoPower}.\n \tShieldPower = #{@shieldPower}.\n \tLoot: #{@loot}.\n \t Damage: #{@damage} "
        end

        def self.newCopy(e)
            new(e.name,e.ammoPower,e.shieldPower,e.loot,e.damage)
        end

        def getUIversion
            EnemyToUI.new(self)
        end

        def fire
            @ammoPower
        end

        def protection
            @shieldPower
        end

        def receiveShot(shot)
           ret = ShotResult::DONOTRESIST
           if protection() >= shot
                ret = ShotResult::RESIST
           end
           ret
        end




        attr_reader :ammoPower, :damage, :loot, :shieldPower , :name 
        
    end

end
